### PR TITLE
Scaffold for Pipeline commands

### PR DESCRIPTION
**Pull request by SoManyHs**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by SoManyHs** [comment]
```
~/workspace/amazon-ecs-archer(pipeline)$ archer pipeline
Command for working with pipelines.

Usage
  archer pipeline [command]

Available Commands
  init        Creates a pipeline for an application.

Flags
  -h, --help   help for pipeline

~/workspace/amazon-ecs-archer(pipeline)$ archer pipeline init --help
Creates a pipeline for an application in a project, using the environments associated with the application."

Usage
  archer pipeline init [flags]

Flags
  -a, --app string                   Name of the application.
  -d, --deploy                       Deploy pipline automatically.
      --enable-cd                    Enables automatic deployment to production environment.
  -e, --environments strings         Environments to add to the pipeline. (default [build])
  -t, --github-access-token string   GitHub personal access token for your GitHub repository.
  -r, --github-repo string           GitHub repository for your application.
  -h, --help                         help for init

Examples
  Create a pipeline for your "frontend".
  $ archer pipeline init \
    --app frontend \
    --github-repo "gitHubUserName/myFrontendApp" \
    --github-access-token file://myGitHubToken \
    --environments stage prod \
    --deploy

```

**Comment by SoManyHs** [comment]
Next up: implementing `Ask()`, `Validate()` and `Execute()`

File: internal/pkg/cli/pipeline_init.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+	"github.com/spf13/afero"
+)
+
+
+// InitPipelineOpts holds the configuration needed to create a new pipeilne
+type InitPipelineOpts struct {
+	// Fields with matching flags.
+	GitHubRepo  string
+	GitHubAccessToken string
+	AppName string
```
Comment (efekarakus) [inline internal/pkg/cli/pipeline_init.go]: Do we need this flag?

I believe we want to pick up all the applications in the workspace.

File: internal/pkg/cli/pipeline_init.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+		Use:   "init",
+		Short: "Creates a pipeline for an application.",
+		Long: `Creates a pipeline for an application in a project, using the environments associated with the application."`,
+		Example: `
+  Create a pipeline for your "frontend".
+  /code $ archer pipeline init \
+    --app frontend \
+    --github-repo "gitHubUserName/myFrontendApp" \
+    --github-access-token file://myGitHubToken \
+    --environments stage prod \
```
Comment (efekarakus) [inline internal/pkg/cli/pipeline_init.go]: :O does this work (separated by spaces), or does it pick up only one of the arguments ("stage"  in this situation)?

File: internal/pkg/cli/pipeline_init.go [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+	cmd := &cobra.Command{
+		Use:   "init",
+		Short: "Creates a pipeline for an application.",
+		Long: `Creates a pipeline for an application in a project, using the environments associated with the application."`,
+		Example: `
+  Create a pipeline for your "frontend".
+  /code $ archer pipeline init \
+    --app frontend \
+    --github-repo "gitHubUserName/myFrontendApp" \
+    --github-access-token file://myGitHubToken \
```
Comment (efekarakus) [inline internal/pkg/cli/pipeline_init.go]: nit: should we give the example with the env var instead (from our last convo with @allisaurus ) `$GITHUB_TOKEN`

File: internal/pkg/cli/pipeline_init.go [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+	return nil
+}
+
+// BuildPipelineInitCmd build the command for creating a new pipeline.
+func BuildPipelineInitCmd() *cobra.Command {
+	opts := &InitPipelineOpts{}
+	cmd := &cobra.Command{
+		Use:   "init",
+		Short: "Creates a pipeline for an application.",
+		Long: `Creates a pipeline for an application in a project, using the environments associated with the application."`,
```
Comment (efekarakus) [inline internal/pkg/cli/pipeline_init.go]: Should this be "Creates a pipeline for the applications in your workspace"?

File: internal/pkg/cli/pipeline_init.go [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+			}
+			return opts.Execute()
+		},
+	}
+	cmd.Flags().StringVarP(&opts.GitHubRepo, "github-repo", "r", "", "GitHub repository for your application.")
+	cmd.Flags().StringVarP(&opts.GitHubAccessToken, "github-access-token", "t", "", "GitHub personal access token for your GitHub repository.")
+	cmd.Flags().StringVarP(&opts.AppName, "app", "a", "", "Name of the application.")
+	cmd.Flags().BoolVarP(&opts.Deploy, "deploy", "d", false, "Deploy pipline automatically.")
+	cmd.Flags().BoolVarP(&opts.EnableCD, "enable-cd", "", false, "Enables automatic deployment to production environment.")
+	cmd.Flags().StringSliceVarP(&opts.Environments, "environments", "e", []string{"build"}, "Environments to add to the pipeline.")
```
Comment (efekarakus) [inline internal/pkg/cli/pipeline_init.go]: Do we need to surface "build" or can that stay hidden?

File: internal/pkg/cli/pipeline_init.go (hunk 1 above)
Comment (SoManyHs) [reply internal/pkg/cli/pipeline_init.go]: Hm, really? I guess that makes sense -- maybe I'm still confused about how we're making this explicit based on the last IRL discussion.

File: internal/pkg/cli/pipeline_init.go (hunk 2 above)
Comment (SoManyHs) [reply internal/pkg/cli/pipeline_init.go]: Oh sorry, this should be comma-separated.

File: internal/pkg/cli/pipeline_init.go (hunk 3 above)
Comment (SoManyHs) [reply internal/pkg/cli/pipeline_init.go]: I thought env var and file were interchangeable for all intents and purposes?

File: internal/pkg/cli/pipeline_init.go (hunk 5 above)
Comment (SoManyHs) [reply internal/pkg/cli/pipeline_init.go]: Hm, not sure. I guess it depends on how we hook it in to @hencrice's transforms.

File: internal/pkg/cli/pipeline_init.go (hunk 3 above)
Comment (efekarakus) [reply internal/pkg/cli/pipeline_init.go]: Yeah, I think you're right, maybe it's more interesting to list out that option

File: internal/pkg/cli/group/group.go [hunk 6]
```diff
@@ -10,4 +10,5 @@ const (
 	GettingStarted = "Getting Started ✨"
 	Develop = "Develop 🔧"
 	Settings = "Settings"
-)
\ No newline at end of file
+	Release = "Release"
```
[the remark this answers is not in the document]
Comment (SoManyHs) [reply internal/pkg/cli/group/group.go]: 🚀 !

File: internal/pkg/cli/group/group.go [hunk 7]
```diff
@@ -10,4 +10,5 @@ const (
 	GettingStarted = "Getting Started ✨"
 	Develop = "Develop 🔧"
 	Settings = "Settings"
-)
\ No newline at end of file
+	Release = "Release 🚀"
```
Comment (kohidave) [inline internal/pkg/cli/group/group.go]: Yay!

File: internal/pkg/cli/pipeline_init.go [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+type InitPipelineOpts struct {
+	// Fields with matching flags.
+	GitHubRepo  string
+	GitHubAccessToken string
+	Deploy bool
+	EnableCD bool
+	Environments []string
+
+	// Interfaces to interact with dependencies.  // TODO
+	fs             afero.Fs
```
Comment (kohidave) [inline internal/pkg/cli/pipeline_init.go]: I don't think we'll need this - the WS should have all the context we need.

File: internal/pkg/cli/pipeline_init.go [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+	cmd := &cobra.Command{
+		Use:   "init",
+		Short: "Creates a pipeline for applications in your workspace.",
+		Long: `Creates a pipeline for the applications in your workspace, using the environments associated with the applications."`,
+		Example: `
+  Create a pipeline for the applications in your workspace
+  /code $ archer pipeline init \
+    --github-repo "gitHubUserName/myFrontendApp" \
+    --github-access-token file://myGitHubToken \
+    --environments "stage,prod" \
```
Comment (kohidave) [inline internal/pkg/cli/pipeline_init.go]: Ah great, so the env order is determined through the order passed in 👍

File: internal/pkg/cli/pipeline_init.go [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+    --deploy`,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			opts.fs = &afero.Afero{Fs: afero.NewOsFs()}
```
Comment (kohidave) [inline internal/pkg/cli/pipeline_init.go]: Probably can get rid of this, too.

File: internal/pkg/cli/pipeline_init.go [hunk 11]
```diff
@@ -0,0 +1,153 @@
...
+			if err != nil {
+				return fmt.Errorf("workspace cannot be created: %w", err)
+			}
+			opts.manifestWriter = ws
+
+			return opts.Validate()
+		},
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			log.Warningln("It's best to run this command in the root of your workspace.")
```
Comment (kohidave) [inline internal/pkg/cli/pipeline_init.go]: We can probably get rid of this warning - only init is best to run from the WS root.

File: internal/pkg/cli/pipeline_init.go [hunk 12]
```diff
@@ -0,0 +1,153 @@
...
+			}
+			if err := opts.Validate(); err != nil { // validate flags
+				return err
+			}
+			return opts.Execute()
+		},
+	}
+	cmd.Flags().StringVarP(&opts.GitHubRepo, "github-repo", "r", "", "GitHub repository for your application.")
+	cmd.Flags().StringVarP(&opts.GitHubAccessToken, "github-access-token", "t", "", "GitHub personal access token for your GitHub repository.")
+	cmd.Flags().BoolVarP(&opts.Deploy, "deploy", "d", false, "Deploy pipline automatically.")
```
Comment (kohidave) [inline internal/pkg/cli/pipeline_init.go]: I can't think of a better description of this - but we should noodle on it a bit. It's like, we're going to actually create the Pipeline after we generate the pipeline.yml for it. Hm - if you can think of something that expresses that in a non-confusing way I'll give you POINTZ

**Pull request merged**